crates/rnote-cli/src/export.rs

```diff
@@ -0,0 +1,673 @@
+                    )
+                }
+                SelectionCommands::All => {
+                    (engine.store.stroke_keys_as_rendered(), "Document is empty")
+                }
+            };
+            if strokes.is_empty() {
+                return Err(anyhow::anyhow!("{err_msg}"));
+            }
+            engine.store.set_selected_keys(&strokes, true);
+            let export_bytes = engine
+                .export_selection(None)
+                .await??
+                .context("No strokes selected")?;
+            store_export(output_file, &export_bytes).await?;
+        }
+        ExportCommands::Doc { .. } => {
+            let Some(export_file_name) = output_file.as_ref().file_name().map(|s| s.to_string_lossy().to_string()) else {
+                return Err(anyhow::anyhow!("Failed to get filename from output_file"));
+            };
+            let export_bytes = engine.export_doc(export_file_name, None).await??;
+            store_export(output_file, &export_bytes).await?;
+        }
+        ExportCommands::DocPages {
+            output_dir,
+            output_file_stem,
+            output_format,
+            ..
+        } => {
+            validators::path_is_dir(output_dir)?;
+            // The output file cannnot be set with this subcommand
+            drop(output_file);
+
+            let export_bytes = engine.export_doc_pages(None).await??;
+            let out_ext = output_format.file_ext();
+            let output_file_stem = match output_file_stem {
+                Some(o) => o.clone(),
+                None => match rnote_file.file_stem() {
+                    Some(stem) => stem.to_string_lossy().to_string(),
+                    None => {
+                        return Err(anyhow::anyhow!(
+                            "Failed to generate output_file_stem from rnote_file"
+                        ))
+                    }
+                },
+            };
+            for (i, bytes) in export_bytes.into_iter().enumerate() {
+                store_export(
+                    &doc_page_output_file(
+                        i,
+                        output_dir,
+                        &out_ext,
+                        &output_file_stem,
+                        on_conflict,
+                        on_conflict_overwrite,
+                    )?,
+                    &bytes,
+                )
+                .await
+                .context(format!(
+                    "Failed to export page {i} of document {}",
+                    rnote_file.display()
+                ))?
+            }
+        }
+    };
+    Ok(())
+}
+
+fn doc_page_output_file(
+    i: usize,
+    output_dir: &Path,
+    out_ext: &str,
+    output_file_stem: &str,
+    on_conflict: OnConflict,
+    on_conflict_overwrite: &mut Option<OnConflict>,
+) -> anyhow::Result<PathBuf> {
+    let mut out = output_dir.join(format!(
+        "{output_file_stem} - page {}{}.{out_ext}",
+        if i + 1 < 10 { "0" } else { "" },
```

Comment:
I believe there is a formatting option for this. It would break if files are in the hundreds or more. If you want to be clean, count the number of files and specify the leading zeros based no that.